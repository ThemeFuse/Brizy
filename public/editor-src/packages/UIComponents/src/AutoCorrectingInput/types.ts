import { Ref } from "react";
import { AutoCorrectingInput } from "./index";

export interface CommonProps {
  className?: string;
  value: number;
  onChange: (value: number) => void;
  min: number;
  max: number;
  step: number;
  onFocus?: VoidFunction;
  onBlur?: VoidFunction;
  onTextChange?: (e: string | number) => void;
  onMouseEnter?: VoidFunction;
  onMouseLeave?: VoidFunction;
  size?: number;
  handleDecrementCb?: () => void;
}

export interface StepperProps extends CommonProps {
  stepperSize: "small" | "medium" | "large" | "auto";
  handleDecrementCb: () => void;
  inputRef: Ref<AutoCorrectingInput>;
}

export interface State {
  text: string | number;
  value: number;
  prevPropsValue: number;
}
