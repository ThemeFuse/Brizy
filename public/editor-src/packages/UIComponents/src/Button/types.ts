import { ReactNode } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  onClick?: VoidFunction;
  reverse?: boolean;
  children?: ReactNode;
  className?: string;
  icon?: IconsName;
  align?: "left" | "center" | "right";
  reverseTheme?: boolean;
  title?: string;
}
