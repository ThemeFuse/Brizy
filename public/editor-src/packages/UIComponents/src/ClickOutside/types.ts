import { ReactElement, RefObject } from "react";

type FunctionalException = (el: HTMLElement) => boolean;

export type Exception = string | FunctionalException;

export type Acc = {
  functionExceptions: FunctionalException[];
  stringExceptions: string[];
};

export interface Props<T extends HTMLElement> {
  children: (ref: RefObject<T>) => ReactElement;
  exceptions: Exception[];
  onClickOutside: (e: MouseEvent) => void;
}
