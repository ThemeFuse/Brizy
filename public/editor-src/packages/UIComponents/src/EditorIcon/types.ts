export enum IconsName {
  Close = "Close",
  CheckSmall = "CheckSmall",
  Spinner = "Spinner",
  PlusCircle = "PlusCircle",
  Duplicate = "Duplicate",
  CheckCircle = "CheckCircle",
  Settings = "Settings",
  Heart = "Heart",
  StreUp = "StreUp",
  StreDown = "StreDown",
  TextAlignCenter = "TextAlignCenter",
  TextAlignLeft = "TextAlignLeft",
  TextAlignRight = "TextAlignRight",
  Circle2 = "Circle2",
  CircleRemove = "CircleRemove",
  "t2-sermon-list" = "MinistryBrandsSermonList",
  "t2-sermon-detail" = "MinistryBrandsSermonDetail",
  "t2-sermon-featured" = "MinistryBrandsSermonFeatured",
  "t2-sermon-layout" = "MinistryBrandsSermonLayout",
  "t2-event-list" = "MinistryBrandsEventList",
  "t2-event-detail" = "MinistryBrandsEventDetail",
  "t2-event-featured" = "MinistryBrandsEventFeatured",
  "t2-event-layout" = "MinistryBrandsEventLayout",
  "t2-event-calendar" = "MinistryBrandsEventCalendar",
  "t2-group-list" = "MinistryBrandsGroupList",
  "t2-group-detail" = "MinistryBrandsGroupDetail",
  "t2-group-featured" = "MinistryBrandsGroupFeatured",
  "t2-group-slider" = "MinistryBrandsGroupSlider",
  "t2-group-layout" = "MinistryBrandsGroupLayout",
  "nc-stre-up" = "StreUp",
  "nc-stre-down" = "StreDown",
  MediaImage = "MediaImage",
  MediaVideo = "MediaVideo",
  MediaMap = "MediaMap",
  Reverse = "Reverse"
}
