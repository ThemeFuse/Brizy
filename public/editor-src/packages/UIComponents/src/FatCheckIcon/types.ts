import { MouseEventHandler } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  icon: IconsName;
  label: string;
  onClick?: VoidFunction;
  onCheck?: VoidFunction;
  isActive?: boolean;
  isChecked?: boolean;
}

export interface CheckMarkProps extends WithClassName {
  isChecked: boolean;
  onClick?: MouseEventHandler;
}
