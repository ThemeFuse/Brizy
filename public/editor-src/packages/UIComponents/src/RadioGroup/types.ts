import { ReactElement } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export type ItemProps = {
  icon: IconsName;
  title?: string;
  value?: string;
  isActive?: boolean;
};

export interface Props extends WithClassName {
  children: ReactElement[];
  onChange: (v: string) => void;
  value?: string;
}
