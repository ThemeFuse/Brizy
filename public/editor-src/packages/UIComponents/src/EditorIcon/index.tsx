import Loadable from "@loadable/component";
import classnames from "classnames";
import React, { CSSProperties, FC, MouseEvent, ReactElement } from "react";
import { IconsName } from "./types";

const LoadableIcon = Loadable(async () => {
  const { Icon } = await import("./components/Icon");
  return Icon;
});
export interface Props {
  className?: string;
  icon: IconsName;
  style?: CSSProperties;
  onClick?: (e: MouseEvent<SVGElement>) => void;
}

export const EditorIcon: FC<Props> = ({
  className,
  icon,
  style,
  onClick
}): ReactElement => {
  const _className = classnames(
    "brz-icon-svg brz-ed-icon-svg align-[initial]",
    className
  );

  return (
    <LoadableIcon
      name={icon}
      className={_className}
      style={style}
      onClick={onClick}
    />
  );
};
