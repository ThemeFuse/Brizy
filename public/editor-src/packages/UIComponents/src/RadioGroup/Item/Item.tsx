import React from "react";
import { EditorIcon } from "../../EditorIcon";
import { ItemProps } from "../types";

export function Item({ icon }: ItemProps) {
  return <EditorIcon icon={icon} />;
}
