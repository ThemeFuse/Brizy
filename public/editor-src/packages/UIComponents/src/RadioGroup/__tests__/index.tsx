import { fireEvent, render, screen } from "@testing-library/react";
import React, { ReactElement } from "react";
import { RadioGroup } from "..";
import { IconsName } from "../../EditorIcon/types";
import { Item } from "../Item/Item";

describe("RadioGroup", () => {
  const onChange = jest.fn();
  const mockChildren: ReactElement[] = [
    <Item
      icon={IconsName.MediaImage}
      isActive={true}
      value="value1"
      title="title1"
      key={1}
    />,
    <Item
      icon={IconsName.MediaImage}
      isActive={false}
      value="value2"
      title="title2"
      key={2}
    />
  ];
  describe("snapshots", () => {
    test("Default usage", () => {
      const group = render(
        <RadioGroup onChange={onChange}>{mockChildren}</RadioGroup>
      );
      expect(group.container.firstChild).toMatchSnapshot();
    });

    test("should render the children", () => {
      render(<RadioGroup onChange={onChange}>{mockChildren}</RadioGroup>);
      const firstChild = screen.getByTitle("title1");
      const secondChild = screen.getByTitle("title2");
      expect(firstChild).toBeInTheDocument();
      expect(secondChild).toBeInTheDocument();
    });

    test("should call onChange when an option is clicked", () => {
      render(<RadioGroup onChange={onChange}>{mockChildren}</RadioGroup>);
      const firstChild = screen.getByTitle("title1");
      const secondChild = screen.getByTitle("title2");
      fireEvent.click(firstChild);
      expect(onChange).toHaveBeenCalledWith("value1");

      fireEvent.click(secondChild);
      expect(onChange).toHaveBeenCalledWith("value2");
    });

    test("should add the active class when the active prop is set to true", () => {
      render(
        <RadioGroup onChange={onChange} value={"value1"}>
          {mockChildren}
        </RadioGroup>
      );
      const firstChild = screen.getByTitle("title1");
      const secondChild = screen.getByTitle("title2");

      expect(firstChild).toHaveClass(
        "brz-ed-control__radio-group2__item--active"
      );
      expect(secondChild).toHaveClass("brz-ed-control__radio-group2__item");
    });
  });
});
