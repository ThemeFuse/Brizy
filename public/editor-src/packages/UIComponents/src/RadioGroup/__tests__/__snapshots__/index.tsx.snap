// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup snapshots Default usage 1`] = `
<div
  class="brz-ed-control__radio-group2"
>
  <div
    class="brz-ed-control__radio-group2__item"
    title="title1"
  />
  <div
    class="brz-ed-control__radio-group2__item"
    title="title2"
  />
</div>
`;
